src/unit.rs

```diff
@@ -163,12 +160,69 @@ impl Unit {
     }
 }
 
-/// Perform a connection. Used recursively for redirects.
+/// Perform a connection. Follows redirects.
 pub(crate) fn connect(
-    unit: Unit,
+    mut unit: Unit,
+    use_pooled: bool,
+    mut body: SizedReader,
+) -> Result<Response, Error> {
+    let mut previous = vec![];
```

Comment:
Nit: I'd like a type annotation on this. Otherwise you have to read pretty far to figure out what's stored in this vec.

Reply:
I'm confused why you say that. `history.push(unit.url.to_string())` is only a couple lines below: https://github.com/algesten/ureq/blob/de669f65eca0d3d5a19b6b750449760e8a50f5ae/src/unit.rs#L172

Reply:
Heh, true. I got hung up on the line before that:

https://github.com/algesten/ureq/blob/de669f65eca0d3d5a19b6b750449760e8a50f5ae/src/unit.rs#L171

And thought "I wonder what is being passed to connect_inner? I guess I'd have to go look at the definition for that function.

In general I've been developing a preference for explicit type annotations in a lot of cases. But I won't be picky about it in this specific spot.